life/src/main.rs

```diff
@@ -0,0 +1,475 @@
+use macroquad::{color::{BLACK, LIGHTGRAY, RED, WHITE}, input::{get_last_key_pressed, is_mouse_button_down, mouse_position, KeyCode}, shapes::{draw_line, draw_rectangle}, text::draw_text, time::get_time, window::{clear_background, next_frame, Conf}};
+
+const ROWS: usize = 3;
+const COLS: usize = 3;
+const OFFSETS: [i8; 3] = [-1, 0, 1];
+const DEAD: bool = false;
+const ALIVE: bool = true;
+const FRAME_DELAY: f64 = 0.3;
+const WINDOW_HEIGHT: i32 = 700;
+const WINDOW_WIDTH: i32 = 800;
+const GRID_HEIGHT: f32 = WINDOW_HEIGHT as f32 - BUTTON_TAB_HEIGHT;
+const GRID_WIDTH: f32 = 800.;
+const BUTTON_TAB_HEIGHT: f32 = 100.;
+const INSTRUCTIONS_TEXT: &str = "To start or stop press 'ENTER'";
+const INSTRUCTIONS_TEXT_X: f32 = 200.;
+const INSTRUCTIONS_TEXT_Y: f32 = 650.;
+const INSTRUCTIONS_TEXT_SIZE: f32 = 30.;
+
+
+/// Verifies if the position represented by the point ('row_i', 'col_i') is inside
+/// the matrix. 
+/// 
+/// # Arguments
+/// 
+/// - `row_i`: Represents the row in the matrix of the point.
+/// - `col_i`: Represents the column in the matrix of the point.
+/// 
+/// # Returns
+/// 
+/// A `bool` indicating if this is true or not.
+fn is_position_valid(row_i: i8, col_i: i8) -> bool {
+    (row_i >= 0 && row_i < ROWS as i8) && (col_i >= 0 && col_i < COLS as i8)
+}
+
+/// Verifies if 2 points on the matrix are the same point. 
+/// 
+/// # Arguments
+/// 
+/// - `row_i_1`: Represents the row in the matrix of point 1.
+/// - `col_i_1`: Represents the column in the matrix of point 1.
+/// - `row_i_2`: Represents the row in the matrix of point 2.
+/// - `col_i_2`: Represents the column in the matrix of point 2.
+/// 
+/// # Returns
+/// 
+/// A `bool` indicating if this is true or not.
+fn is_different_cell(row_i_1: i8, col_i_1: i8, row_i_2: i8, col_i_2: i8) -> bool {
+    row_i_1 != row_i_2 || col_i_1 != col_i_2
+}
+
+/// Changes the state of specific cells in the matrix.
+/// 
+/// # Arguments
+/// 
+/// - `cells`: A Vec which contains tuples that represent the cells in the amtrix to modify.
+/// - `state`: Represents wether the cell is alive or dead.
+/// - `matrix`: An array with arrays that represent the matrix which contains every cell.
+fn manage_cell_state(cells: Vec<(usize, usize)>, state: bool, matrix: &mut [[bool; COLS]; ROWS]) {
```

Comment:
Here is another somewhat advanced idea, makign use of rust traits:

- Change cells to be a Vec<Cell>, where cell is a struct with 2 fields like here.
- Make a matrix struct holding the matrix array.
- Implement the index trait for the matrix struct using Cell: https://doc.rust-lang.org/std/ops/trait.Index.html
- 
```rust
impl Index<Cell> for Matrix {
    type Output = usize;

    fn index(&self, cell: Cell) -> &Self::Output {
        &matrix.0[cell.y][cell.x] // for example, just to show
    }
}
```

To allow mutating u also need https://doc.rust-lang.org/std/ops/trait.IndexMut.html